server/src/main/java/org/elasticsearch/snapshots/SnapshotShardsService.java

```diff
@@ -282,9 +283,9 @@ private void startNewShards(SnapshotsInProgress.Entry entry, Map<ShardId, IndexS
                 final IndexShardSnapshotStatus snapshotStatus = shardEntry.getValue();
                 final IndexId indexId = indicesMap.get(shardId.getIndexName());
                 assert indexId != null;
-                assert entry.useShardGenerations() || snapshotStatus.generation() == null :
+                assert entry.version().onOrAfter(SnapshotsService.SHARD_GEN_IN_REPO_DATA_VERSION) || snapshotStatus.generation() == null :
```

Comment:
Instead of the `onOrAfter` checks, I would prefer a method on `SnapshotsService` of the form

```
boolean useShardGenerations(Version repositoryMetaVersion) {
  return repositoryMetaVersion.onOrAfter(Version.V_7_6_0);
}
```

and then remove the public `SHARD_GEN_IN_REPO_DATA_VERSION` field.

Reply:
Added the method and dried up all spots that it applied to. I'm afraid we'll have to keep the constant though so it can be written to the wire in the BwC case in `SnapshotsInProgress.Entry`.
We can get rid of it in `master` though once `7.7` is out :)